module/zfs/dbuf.c

```diff
@@ -1973,6 +1973,75 @@ dbuf_redirty(dbuf_dirty_record_t *dr)
 	}
 }
 
+dbuf_dirty_record_t *
+dbuf_dirty_lightweight(dnode_t *dn, uint64_t blkid, dmu_tx_t *tx)
+{
+	rw_enter(&dn->dn_struct_rwlock, RW_READER);
+	ASSERT(!dn->dn_objset->os_raw_receive ||
```

Comment:
[nit] personally I feel like IMPLY(os_raw_receive, maxblkid >= blkid) is slightly clearer than this?

Reply:
I agree, changed.